### PR TITLE
Add support to debug OIDC requests & responses

### PR DESCRIPTION
**Pull request by Sathiyakugan**

### CODE REVIEW

File: modules/identity-java-agent/java-agent/src/main/java/org/wso2/carbon/identity/java/agent/config/InterceptorConfigReader.java [hunk 1]
```diff
@@ -36,7 +36,7 @@
     public List<InterceptorConfig> readConfig() {
 
         ArrayList<InterceptorConfig> result = new ArrayList<>();
-
+        // TODO Add 2 private methods
```
Comment (darshanasbg) [inline modules/identity-java-agent/java-agent/src/main/java/org/wso2/carbon/identity/java/agent/config/InterceptorConfigReader.java:39]: Please address the TODO

File: modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/dap/serializer/VariableEncoderRegistry.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import org.wso2.carbon.identity.developer.lsp.debug.dap.serializer.encoders.OIDCAuthzRequestEncoder;
+import org.wso2.carbon.identity.developer.lsp.debug.dap.serializer.encoders.OIDCTokenRequestEncoder;
+import org.wso2.carbon.identity.developer.lsp.debug.dap.serializer.encoders.OIDCTokenResponseEncoder;
+import org.wso2.carbon.identity.developer.lsp.debug.dap.serializer.encoders.SAMLRequestEncoder;
+import org.wso2.carbon.identity.developer.lsp.debug.dap.serializer.encoders.SAMLResponseEncoder;
+import org.wso2.carbon.identity.developer.lsp.debug.dap.serializer.encoders.VariableEncoder;
+
+import java.util.HashMap;
+
+public class VariableEncoderRegistry {
```
Comment (darshanasbg) [inline modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/dap/serializer/VariableEncoderRegistry.java:33]: Add class comments

File: modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java [hunk 3]
```diff
@@ -206,9 +206,9 @@ private void sendRequestToClient(Session websocketSession, ProtocolMessage messa
     private Map.Entry<Session, DebugSession> findInterestedDebugSession(MethodContext methodContext) {
 
         //For not, just return the first entry. We need to have a better filter later.
-        if (!methodContext.getClassName().equals("org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet")) {
-            return null;
-        }
+//        if (!methodContext.getClassName().equals("org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet")) {
```
Comment (darshanasbg) [inline modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java:209]: Why commented code?

File: pom.xml [hunk 4]
```diff
@@ -172,6 +172,11 @@
                 <version>${org.powermock.version}</version>
                 <scope>test</scope>
             </dependency>
+            <dependency>
+                <groupId>com.auth0</groupId>
```
Comment (darshanasbg) [inline pom.xml:176]: use commonly used jwt library across the platform

File: modules/identity-java-agent/java-agent/src/main/java/org/wso2/carbon/identity/java/agent/config/InterceptorConfigReader.java (hunk 1 above)
Comment (Sathiyakugan) [reply modules/identity-java-agent/java-agent/src/main/java/org/wso2/carbon/identity/java/agent/config/InterceptorConfigReader.java]: Fixed

File: modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/dap/serializer/VariableEncoderRegistry.java (hunk 2 above)
Comment (Sathiyakugan) [reply modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/dap/serializer/VariableEncoderRegistry.java:36]: Fixed

File: modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java (hunk 3 above)
Comment (Sathiyakugan) [reply modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java]: Fixed .

File: pom.xml (hunk 4 above)
Comment (Sathiyakugan) [reply pom.xml]: Fixed .

File: modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java [hunk 5]
```diff
@@ -205,10 +205,7 @@ private void sendRequestToClient(Session websocketSession, ProtocolMessage messa
 
     private Map.Entry<Session, DebugSession> findInterestedDebugSession(MethodContext methodContext) {
 
-        //For not, just return the first entry. We need to have a better filter later.
-        if (!methodContext.getClassName().equals("org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet")) {
-            return null;
-        }
+        // Have to add mechanism to corresponding the relevant tenant
```
Comment (darshanasbg) [inline modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java:208]: Lets create a issue to track this.

File: modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/oidc/builders/OIDCAuthzRequestVariable.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+import org.wso2.carbon.identity.developer.lsp.debug.DAPConstants;
+
+import java.util.HashMap;
+
+/**
+ * Variables Holds the necessary Variables for OAUTH Authorization Request.
+ */
+public class OIDCAuthzRequestVariable implements OIDCAuthzRequestVariablePlan {
+    // TODO Change to proper name.
```
Comment (darshanasbg) [inline modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/oidc/builders/OIDCAuthzRequestVariable.java:29]: Address this one.

Comment (Sathiyakugan) [reply modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/oidc/builders/OIDCAuthzRequestVariable.java]: Fixed

File: modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java (hunk 5 above)
Comment (Sathiyakugan) [reply modules/server-endpoints/language-server/src/main/java/org/wso2/carbon/identity/developer/lsp/debug/runtime/DebugSessionManagerImpl.java:208]: Fixed

**Comment by Sathiyakugan** [comment]
closes https://github.com/wso2/product-is/issues/8286

**Comment by Sathiyakugan** [comment]
closes https://github.com/wso2/product-is/issues/8287

**Pull request merged**